docs/wiki/99 - Developers/Code Review Guide.md

```diff
@@ -0,0 +1,49 @@
+# Code Review Guide
+
+## Creating a good pull request
+
+The [CONTRIBUTING](../../../CONTRIBUTING.md) document on GitHub explains how to author a good pull requests.
+If you follow these guidelines, we'll be better able to review and merge your code.
+
+## The Code Review Process
+When our team reviews pull requests we are evaluating whether the changed code
+accomplishes its intentions, provides a good basis for future work, and correponds
+to our style guidelines.
+
+### When can a pull request be merged?
+We require 3 things to be true before a pull request can be merged:
+
+1. The code must pass our automated tests.
+2. The code must be manually tested by a member of our team. When the code has
+been confirmed by manual tests, the reviewer should add the `Passed Manual Tests`
+label on GitHub.
+3. The pull request must have an 'Approved' review status.
+
+This document will help you to understand how to get your code to be Approved.
+
+## Checklist 
+
+In addition to what's listed in the CONTRIBUTING document, here are some things we are
+looking for when deciding whether to Approve your pull request.
+
+- [ ] 1.  Do database calls use prepared statements?
+- [ ] 2.  Have new tests been added?
+- [ ] 3.  Is CSS/Javascript/PHP/HTML code separated into their own files?
+- [ ] 4.  Have database changes been reflected in the schema, a new patch, and
+in Raisinbread?
+- [ ] 5.  Are the comments reasonable/understandable?
+- [ ] 6.  Are the variable names meaningful? In the proper scope?
+- [ ] 7.  Does the logic/structure make sense? Is it as efficient as possible?
+- [ ] 8.  Are inheritance/classes/etc used appropriately?
+- [ ] 9.  Is there an effort to limit code duplication?
+- [ ] 10. Are functions concise and clear?
+- [ ] 11. Are the commit messages meaningful, atomic, and rebased/squashed into a clean tree?
+- [ ] 12. Is the pull request appropriately named (starting with Module name or Core) and appropriately labelled?
+- [ ] 13. Is the comment on the pull request meaningful and informative, even to someone with limited knowledge of the code?
+- [ ] 14. Are there any other implications for existing projects that need to be documented or addressed? 
+- [ ] 15. Are changes made in a backward compatible manner (where possible)?
+- [ ] 16. Has relevant documentation been updated?
+- [ ] 17. Is the code written with thought given to how it can be maintained and extended? Is it written elegantly, without "hacks"?
```

Comment:
- Does it use the `handle` function and not an `ajax` directory?
- Where possible, is new frontend code written using React rather than smarty templates?
- Are URLs structured as a well-designed API which follows the HTTP specification, rather than a single magic endpoint that changes behaviour via a multitude of query parameters?

Reply:
> Are URLs structured as a well-designed API which follows the HTTP specification, rather than a single magic endpoint that changes behaviour via a multitude of query parameters?

I agree but I feel like very few people are going to understand what this sentence actually means. We might need a more in-depth document that gets into the nitty-gritty of LORIS PHP guts.

Reply:
Once written, this document should be referenced here!

Reply:
Whether they're going to understand the comment or not (I don't care if it's rewritten), it's the most common problem that prevents modules from getting merged and takes a long time to resolve, so it should be here in some form and people should be thinking about it before it gets to code review.

Reply:
I added a brief note here. Maybe this point can be added in more detail to #6048 because this is an important point for people submitting new modules.